File: app/src/main/java/info/nightscout/androidaps/plugins/constraints/objectives/objectives/Objective5.java
package info.nightscout.androidaps.plugins.constraints.objectives.objectives;

import java.util.List;

import javax.inject.Inject;

import dagger.android.HasAndroidInjector;
import info.nightscout.androidaps.R;
import info.nightscout.androidaps.interfaces.Constraint;
import info.nightscout.androidaps.plugins.constraints.safety.SafetyPlugin;
import info.nightscout.androidaps.utils.T;

public class Objective5 extends Objective {
    @Inject SafetyPlugin safetyPlugin;

    public Objective5(HasAndroidInjector injector) {
        super(injector, "maxiobzero", R.string.objectives_maxiobzero_objective, R.string.objectives_maxiobzero_gate);
    }

    @Override
    protected void setupTasks(List<Task> tasks) {
        tasks.add(new MinimumDurationTask(T.days(5).msecs()));
        tasks.add(new Task(R.string.closedmodeenabled) {
            @Override
            public boolean isCompleted() {
                Constraint<Boolean> closedLoopEnabled = new Constraint<>(true);
                safetyPlugin.isClosedLoopAllowed(closedLoopEnabled);
                return closedLoopEnabled.value();
            }
        });
    }
}

File: plugins/constraints/src/main/java/info/nightscout/plugins/constraints/objectives/objectives/Objective5.kt
package info.nightscout.plugins.constraints.objectives.objectives

import dagger.android.HasAndroidInjector
import info.nightscout.core.constraints.ConstraintObject
import info.nightscout.plugins.constraints.R
import info.nightscout.plugins.constraints.safety.SafetyPlugin
import info.nightscout.rx.logging.AAPSLogger
import info.nightscout.shared.utils.T
import javax.inject.Inject

@Suppress("SpellCheckingInspection")
class Objective5(injector: HasAndroidInjector) : Objective(injector, "maxiobzero", R.string.objectives_maxiobzero_objective, R.string.objectives_maxiobzero_gate) {

    @Inject lateinit var safetyPlugin: SafetyPlugin
    @Inject lateinit var aapsLogger: AAPSLogger

    init {
        tasks.add(MinimumDurationTask(this, T.days(5).msecs()))
        tasks.add(
            object : Task(this, R.string.closedmodeenabled) {
                override fun isCompleted(): Boolean {
                    val closedLoopEnabled = ConstraintObject(false, aapsLogger)
                    safetyPlugin.isClosedLoopAllowed(closedLoopEnabled)
                    return closedLoopEnabled.value()
                }
            }.learned(Learned(R.string.objectives_maxiobzero_learned))
        )
    }
}